ambassador/ambassador/mapping.py

```diff
@@ -122,6 +122,46 @@ def get(self, key, *args):
         else:
             return self.attrs.get(key)
 
+    @staticmethod
+    def add_to_dict_if_key_exists(key1, dict1, key2, dict2):
+        """If dict1[key1] exists, and
```

Comment:
This should probably be something like `cors_save_element`. That's semantically narrow enough that it's OK for it to modify `self.cors` instead of having to have a dict passed in. 

```python
def cors_save_element(self, cors_key, params, param_key):
    ...
    self.cors[cors_key] = params[param_key] # or whatever
    ...
```

and, later, 

```python
self.cors_save_element('credentials', route_cors, 'allow_credentials')
```

Reply:
makes sense, fixed